File: javascript/src/models/enums.js
// FleetLink domain enums (Module 2.C). Values match the FSD §3 exactly and are stored/exposed as
// strings (root CLAUDE.md: "enums are stored and exposed as strings", never magic integers).
export const VehicleStatus = Object.freeze(['Active', 'InMaintenance', 'Retired']);
export const DriverStatus = Object.freeze(['Active', 'Inactive']);
export const WorkOrderType = Object.freeze(['Scheduled', 'Inspection', 'Breakdown']);
export const WorkOrderPriority = Object.freeze(['Low', 'Medium', 'High', 'Critical']);
export const WorkOrderStatus = Object.freeze(['Open', 'InProgress', 'OnHold', 'Completed', 'Cancelled']);

// Convenience set-membership check used by the model factories (boundary shape only — the business
// rules that USE these enums, e.g. the status state machine in FSD §5.10, arrive in Module 2.D).
export const isMember = (allowed, value) => allowed.includes(value);

File: dotnet/src/FleetLink.Api/Models/Enums.cs
namespace FleetLink.Api.Models;

// FleetLink domain enums — FSD §3. Stored and exposed as STRINGS (root CLAUDE.md), never magic
// integers; the string mapping is applied when the model is persisted (Module 2.D).
public enum VehicleStatus { Active, InMaintenance, Retired }
public enum DriverStatus { Active, Inactive }
public enum WorkOrderType { Scheduled, Inspection, Breakdown }
public enum WorkOrderPriority { Low, Medium, High, Critical }
public enum WorkOrderStatus { Open, InProgress, OnHold, Completed, Cancelled }
